lib/typing_env.mli

```diff
@@ -35,11 +35,12 @@ val expand_tconstr :
     the typing environment
 *)
 
-val fully_expand_tconstr :
+val expansion_lst :
   typing_env:t ->
+  type_expr:Types.type_expr ->
   path:Path.t ->
   args:Types.type_expr list ->
-  Types.type_expr option
+  (Types.type_expr * Types.type_declaration option) list
```

Comment:
This needs to be clarified a bit, I'm not sure how to interpret the option and I don't think we need the whole type declaration either.